test/metadata_test.go

```diff
@@ -98,23 +96,23 @@ func (s) TestInvalidMetadata(t *testing.T) {
 
 		ctx = metadata.NewOutgoingContext(ctx, test.md)
 		if _, err := ss.Client.EmptyCall(ctx, &testpb.Empty{}); !reflect.DeepEqual(test.want, err) {
-			t.Fatalf("call ss.Client.EmptyCall() validate metadata which is %v got err :%v, want err :%v", test.md, err, test.want)
+			t.Errorf("call ss.Client.EmptyCall() validate metadata which is %v got err :%v, want err :%v", test.md, err, test.want)
 		}
 	}
 
-	for range tests {
+	// call the stream server's api to drive the server-side unit testing
+	for _, test := range tests {
 		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
 		stream, err := ss.Client.FullDuplexCall(ctx, grpc.WaitForReady(true))
```

Comment:
Is the `grpc.WaitForReady` necessary here?  I suspect it shouldn't be since we shouldn't get a connection error AFAIK.

(Unless sending illegal trailers will lead to a connection error & loss, in which case that is a behavior we should fix later.)